src/pages/Signup/index.tsx

```diff
@@ -0,0 +1,128 @@
+import { useAppDispatch, useAppSelector } from 'app/hooks';
+import { Header } from 'components';
+import { clearError, setFormError, signUpUser } from 'features/User/userSlice';
+import { useEffect, useState } from 'react';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+import { validateEmail } from 'utils';
+
+export function Signup() {
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const dispatch = useAppDispatch();
+  const { user, error } = useAppSelector((state) => state.user);
+  const navigate = useNavigate();
+  const location: any = useLocation();
```

Comment:
suggestion: avoid using any, you will not learn typescript if you start using any and point of using typescript not met